From 9ad80afc0c79311c7d67b4dd67aa80a41e039263 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 28 Apr 2026 09:36:18 +0100
Subject: [PATCH] daemon: Move some deprecated functions to new Luks module

No change, just code movement.

(cherry picked from commit 5e7aef40c2e9a81f5ad3f8c2e1ff1389e7825efb)
---
 .gitignore                           |  1 +
 daemon/Makefile.am                   |  3 +++
 daemon/cryptsetup.ml                 |  9 +-------
 daemon/luks.ml                       | 32 ++++++++++++++++++++++++++++
 generator/actions_core_deprecated.ml |  6 +++---
 5 files changed, 40 insertions(+), 11 deletions(-)
 create mode 100644 daemon/luks.ml

diff --git a/.gitignore b/.gitignore
index c8a8312fa..7715d2cfc 100644
--- a/.gitignore
+++ b/.gitignore
@@ -95,6 +95,7 @@ Makefile.in
 /daemon/ldm.mli
 /daemon/link.mli
 /daemon/listfs.mli
+/daemon/luks.mli
 /daemon/lvm.mli
 /daemon/lvm_dm.mli
 /daemon/lvm_full.mli
diff --git a/daemon/Makefile.am b/daemon/Makefile.am
index d4a805046..6f3b9e7d5 100644
--- a/daemon/Makefile.am
+++ b/daemon/Makefile.am
@@ -49,6 +49,7 @@ generator_built = \
 	ldm.mli \
 	link.mli \
 	listfs.mli \
+	luks.mli \
 	lvm.mli \
 	lvm_dm.mli \
 	lvm_full.mli \
@@ -296,6 +297,7 @@ SOURCES_MLI = \
 	ldm.mli \
 	link.mli \
 	listfs.mli \
+	luks.mli \
 	lvm.mli \
 	lvm_dm.mli \
 	lvm_full.mli \
@@ -327,6 +329,7 @@ SOURCES_ML = \
 	blkid.ml \
 	btrfs.ml \
 	cryptsetup.ml \
+	luks.ml \
 	devsparts.ml \
 	file_helper.ml \
 	file.ml \
diff --git a/daemon/cryptsetup.ml b/daemon/cryptsetup.ml
index 6f677aef0..002e0993a 100644
--- a/daemon/cryptsetup.ml
+++ b/daemon/cryptsetup.ml
@@ -1,5 +1,5 @@
 (* guestfs-inspection
- * Copyright (C) 2009-2025 Red Hat Inc.
+ * Copyright (C) 2009-2026 Red Hat Inc.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -70,10 +70,3 @@ let cryptsetup_close device =
   ignore (command "cryptsetup" ["close"; mapname]);
 
   udev_settle ()
-
-(* Deprecated APIs for backwards compatibility. *)
-let luks_open device key mapname =
-  cryptsetup_open ~crypttype:"luks" device key mapname
-let luks_open_ro device key mapname =
-  cryptsetup_open ~crypttype:"luks" ~readonly:true device key mapname
-let luks_close = cryptsetup_close
diff --git a/daemon/luks.ml b/daemon/luks.ml
new file mode 100644
index 000000000..e7e6deefe
--- /dev/null
+++ b/daemon/luks.ml
@@ -0,0 +1,32 @@
+(* guestfs-inspection
+ * Copyright (C) 2009-2026 Red Hat Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *)
+
+open Printf
+open Unix
+
+open Std_utils
+
+open Utils
+
+(* Deprecated APIs for backwards compatibility. *)
+let luks_open device key mapname =
+  Cryptsetup.cryptsetup_open ~crypttype:"luks" device key mapname
+let luks_open_ro device key mapname =
+  Cryptsetup.cryptsetup_open ~crypttype:"luks" ~readonly:true device key mapname
+let luks_close =
+  Cryptsetup.cryptsetup_close
diff --git a/generator/actions_core_deprecated.ml b/generator/actions_core_deprecated.ml
index 308495c45..d930b8b88 100644
--- a/generator/actions_core_deprecated.ml
+++ b/generator/actions_core_deprecated.ml
@@ -843,7 +843,7 @@ physical volumes, volume groups and logical volumes." };
   { defaults with
     name = "luks_open"; added = (1, 5, 1);
     style = RErr, [String (Device, "device"); String (Key, "key"); String (PlainString, "mapname")], [];
-    impl = OCaml "Cryptsetup.luks_open";
+    impl = OCaml "Luks.luks_open";
     optional = Some "luks";
     deprecated_by = Replaced_by "cryptsetup_open";
     shortdesc = "open a LUKS-encrypted block device";
@@ -869,7 +869,7 @@ devices.|} };
   { defaults with
     name = "luks_open_ro"; added = (1, 5, 1);
     style = RErr, [String (Device, "device"); String (Key, "key"); String (PlainString, "mapname")], [];
-    impl = OCaml "Cryptsetup.luks_open_ro";
+    impl = OCaml "Luks.luks_open_ro";
     optional = Some "luks";
     deprecated_by = Replaced_by "cryptsetup_open";
     shortdesc = "open a LUKS-encrypted block device read-only";
@@ -880,7 +880,7 @@ mapping is created." };
   { defaults with
     name = "luks_close"; added = (1, 5, 1);
     style = RErr, [String (Device, "device")], [];
-    impl = OCaml "Cryptsetup.luks_close";
+    impl = OCaml "Luks.luks_close";
     optional = Some "luks";
     deprecated_by = Replaced_by "cryptsetup_close";
     shortdesc = "close a LUKS device";
-- 
2.47.3

